aiohttp/connector.py

```diff
@@ -978,6 +1001,60 @@ def _get_fingerprint_and_hashfunc(self, req):
 
         return transport, proto
 
+    async def _create_socks_proxy_connection(self, req):
+        if aiosocks is None:
+            raise RuntimeError(
+                "{} requires aiosocks library".format(req.proxy.scheme))
+
+        sslcontext = self._get_ssl_context(req)
+
+        if not req.socks_remote_resolve:
+            try:
+                dst_hosts = list(await self._resolve_host(req.host, req.port))
+                dst = dst_hosts[0]['host'], dst_hosts[0]['port']
+            except OSError as exc:
+                raise ClientConnectorError(
+                    req.connection_key, exc) from exc
+        else:
+            dst = req.host, req.port
+
+        try:
+            proxy_hosts = await self._resolve_host(
+                req.proxy.host, req.proxy.port)
+        except OSError as exc:
+            raise ClientConnectorError(
+                req.connection_key, exc) from exc
+
+        last_exc = None
+
+        for hinfo in proxy_hosts:
+            if req.proxy.scheme == 'socks4':
+                proxy = aiosocks.Socks4Addr(hinfo['host'], hinfo['port'])
+            else:
+                proxy = aiosocks.Socks5Addr(hinfo['host'], hinfo['port'])
+
+            try:
+                transp, proto = await self._wrap_create_connection(
+                    self._factory, proxy, req.proxy_auth, dst,
+                    loop=self._loop, remote_resolve=req.socks_remote_resolve,
+                    ssl=sslcontext, family=hinfo['family'],
+                    proto=hinfo['proto'], flags=hinfo['flags'],
+                    local_addr=self._local_addr, req=req,
+                    client_error=ClientProxyConnectionError,
+                    server_hostname=req.host if sslcontext else None)
+            except ClientConnectorError as exc:
+                last_exc = exc
+                continue
+
+            try:
+                self._check_fingerprint(req, transp, req.host, req.port)
```

Comment:
The connection is proxies by SOCKS server.
Certeficate, if any, is the SOCKS certificate (I doubt if SOCKS specifies the mode at all).
Why do fingerprint checking?
Fingerprints exist for checking a real server, not proxy.

Reply:
To connect with socks server `aiosocks` not use the certificate.
First, communication with socks server, then start tls with destination server.

Reply:
Fingerprint is checked for transport's certificate.
Transport is a connection to SOCKS server.
Thus the check is performed for SOCKS server cert, not proxied host.
Please correct me if I'm wrong.

Reply:
Pretty much.  Essentially there's only two possibilities; either:

 1. The TLS/SSL check will be performed on a cert for the proxy server, assuming there is one; or
 
 2. The target server will report what the TLS credentials should be, but the presence of the proxy will trigger a MitM alert of some kind.
 
The latter is most commonly seen with improperly configured transparent proxies.